src/Components/Blazor/Server/src/Services/BlazorHostingServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Net.Mime;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.StaticFiles;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class BlazorHostingServiceCollectionExtensions
+    {
+        public static IServiceCollection AddBlazorStaticFilesConfiguration(this IServiceCollection services)
+        {
+            services.Configure<StaticFileOptions>(options =>
+            {
+            });
+
+            services.TryAddEnumerable(ServiceDescriptor.Singleton<IConfigureOptions<StaticFileOptions>, ClientSideBlazorStaticFilesConfiguration>());
+
+            return services;
+
+        }
+
+        private class ClientSideBlazorStaticFilesConfiguration : IConfigureOptions<StaticFileOptions>
```

Comment:
separate file + tests

Reply:
This is going away in a future PR

Reply:
Same. Can we file an issue to clean this up?

Reply:
I'm using the same issue https://github.com/dotnet/aspnetcore/issues/19060